grpc.go

```diff
@@ -591,6 +620,25 @@ func (a *agentGRPC) CreateContainer(ctx context.Context, req *pb.CreateContainer
 		return emptyResp, err
 	}
 
+	if a.sandbox.guestHooksPresent {
+		// Add any custom OCI hooks to the spec
+		a.sandbox.addGuestHooks(ociSpec)
+
+		// write the OCI spec to a file so that hooks can read it
+		err = writeSpecToFile(ociSpec)
+		if err != nil {
+			return emptyResp, err
+		}
+
+		// Change cwd because libcontainer assumes the bundle path is the cwd:
+		// https://github.com/opencontainers/runc/blob/v1.0.0-rc5/libcontainer/specconv/spec_linux.go#L157
+		oldcwd, err := changeToBundlePath(ociSpec)
```

Comment:
It's worth noting that the code piece works because golang sets CLONE_FS when creating new M (https://github.com/golang/go/blob/2595fe7fb6f272f9204ca3ef0b0c55e66fb8d90f/src/runtime/os_linux.go#L131) so CWD is shared among all os threads.